test/e2e/storage/testsuites/provisioning.go

```diff
@@ -327,6 +327,26 @@ func (p *provisioningTestSuite) DefineTests(driver storageframework.TestDriver,
 		}
 		wg.Wait()
 	})
+
+	ginkgo.It("should mount multiple PV pointing to the same storage on the same node", func() {
+		// csi-hostpath driver does not support this test case. In this test case, we have 2 PV containing the same underlying storage.
+		// during the NodeStage call for the second volume, csi-hostpath fails the call, because it thinks the volume is already staged at a different path.
+		// Note: This is not an issue with driver like PD CSI where the NodeStage is a no-op for block mode.
```

Comment:
This test is also failing for PMEM-CSI. The driver fails with:

```
pmem-csi-intel-com-node-5hr56/pmem-driver@pmem..ker1: 	rpc error: code = Internal desc = mount filesystem failed: "/bin/mount -c -o bind /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pv-sxlkj/globalmount /var/lib/kubelet/pods/c4f6bd6f-0414-4ce2-8c2b-1c4efb637c58/volumes/kubernetes.io~csi/pv-sxlkj/mount": command failed: exit status 32
pmem-csi-intel-com-node-5hr56/pmem-driver@pmem..ker1: 	Combined stderr/stdout output: mount: /var/lib/kubelet/pods/c4f6bd6f-0414-4ce2-8c2b-1c4efb637c58/volumes/kubernetes.io~csi/pv-sxlkj/mount: special device /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pv-sxlkj/globalmount does not exist.
pmem-csi-intel-com-node-5hr56/pmem-driver@pmem..ker1:  >
```

I don't understand this comment here. Is it about the if check (next line)? Or for the entire test? Then it should come before "ginkgo.It".

About the "csi-hostpath driver does not support this test case" - does that mean that the test is making assumptions about the CSI driver, i.e. not all CSI drivers are not required to pass it? Then the test should be opt-in instead of opt-out.

Reply:
> kubernetes.io/csi/pv/pv-sxlkj/globalmount does not exist.

This is odd. Point of this PR was to replace PV name in the global mount point path with `sha(volumeHandle)`. If two PVs share the same handle, they should share the global mount.
Use case: a volume that is shared into multiple namespaces and thus must have PV+PVC for each of them.

To be honest, I don't know why the test is skipped for block volumes.

Reply:
Does your test match Kubernetes version? I can imagine this can fail with older kubernetes.

Reply:
No, I was running it against Kubernetes 1.22. So the test fails with older Kubernetes because older kubelet didn't properly support this scenario? Re-reading the feature PR description that seems to be the case.

In a better world, the test would have a comment explaining this. As it stands now, the name of the test is "CSI driver XYZ should mount multiple PV pointing to the same storage on the same node" while in reality it is "kubelet should mount multiple PV pointing to the same storage on the same node".

But that doesn't explain the special exception for block devices and csi-driver-host-path.

Reply:
this change is merged for 1.24, so I think need to test your csi driver against kubernetes 1.24?